share/wake/lib/system/io.wake

```diff
@@ -187,6 +193,16 @@ export def write (path: String) (content: String): Result Path Error =
 
     writeImp (parent, Nil) 0644 spath content
 
+# Writes a string to an intermediate file that will not conflict with other files.
+#
+# Idential prefix+files will be automatically deduplicated
+export def writeTempFile (prefix: String) (content: String): Result Path Error =
+    def name = "{prefix}.{hashString content}"
+
+    require Pass dir = mkdir "build/.temp"
```

Comment:
This should be `.build`

Reply:
I feel like `tmp` is a more common name for these directories but that obviously doesn't matter.

Reply:
To avoid nesting dotfiles, maybe `.build/tmp`?

Reply:
Thinking about it further, though, `tmp` seems like it's a bit underspecified because it *is* such a common name; `.build/temp-writes`?  That doesn't really flow nicely.

Reply:
`.build/tmp/writes`?

Reply:
Yeah, that seems good.  Allows further temporary namespacing if anything comes up in the future.